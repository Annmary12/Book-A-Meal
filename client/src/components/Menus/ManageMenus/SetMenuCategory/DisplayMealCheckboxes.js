import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import styles from '../set-menu.scss';

/**
 * @class DisplayMealCheckboxes
 */
class DisplayMealCheckboxes extends Component {
  handleCheck = meal => () => {
    const { toggleMealInCategory } = this.props;
    toggleMealInCategory(meal);
  }

  /**
   * @returns {JSX} React JSX
   */
  render() {
    const {
      meals,
      categoryMealIds,
      errors,
      categoryIndex
    } = this.props;

    const error = errors[`categories[${categoryIndex}].meals`];

    return (
      <div>
        <b>Meals:</b>
        <div
          className={classnames({
            [styles.meals]: true,
            [styles.error]: error
          })}
        >
          {
            meals.map(meal => (
              <div key={meal.id}>
                <label className={styles.checkBoxLabel}>
                  {meal.title}
                  <input
                    type="checkbox"
                    checked={categoryMealIds.includes(meal.id) ? 'checked' : ''}
                    onChange={this.handleCheck(meal)}
                  />
                  <span className={styles.checkmark} />
                </label>
              </div>
            ))
          }
        </div>
        {
        error &&
        <div className={styles.mealsErrorMessage}>{error}</div>
      }
      </div>
    );
  }
}

DisplayMealCheckboxes.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  toggleMealInCategory: PropTypes.func.isRequired,
  categoryMealIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  errors: PropTypes.objectOf(PropTypes.string).isRequired,
  categoryIndex: PropTypes.number.isRequired
};

export default DisplayMealCheckboxes;
