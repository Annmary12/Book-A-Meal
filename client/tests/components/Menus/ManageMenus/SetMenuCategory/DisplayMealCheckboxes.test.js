import React from 'react';
import DisplayMealCheckboxes from '../../../../../src/components/Menus/ManageMenus/SetMenuCategory/DisplayMealCheckboxes';

const props = {
  meals: [],
  toggleMealInCategory: jest.fn(),
  categoryMealIds: [1],
  errors: {'categories[0].meals': 'error'},
  categoryIndex: 0
};

describe('DisplayMealCheckboxes component', () => {
  it('should match snapshot if error exists', () => {
    const wrapper = shallow(
      <DisplayMealCheckboxes {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot if error doesnt exists', () => {
    const wrapper = shallow(
      <DisplayMealCheckboxes
        {...props}
        errors={{}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if meals are present', () => {
    const wrapper = shallow(
      <DisplayMealCheckboxes
        {...props}
        meals={[
          { id: 1, title: 'rice'},
          { id: 2, title: 'beans'}
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('handleCheck method', () => {
    const wrapper = shallow(
      <DisplayMealCheckboxes
        {...props}
        meals={[
          { id: 1, title: 'rice'},
          { id: 2, title: 'beans'}
        ]}
      />
    );
    it('should call toggleMealInCategory prop function', () => {
      const meal = { id: 1, title: 'rice'};
      wrapper.instance().handleCheck(meal)();

      expect(props.toggleMealInCategory).toBeCalledWith(meal);
    });
  });
});