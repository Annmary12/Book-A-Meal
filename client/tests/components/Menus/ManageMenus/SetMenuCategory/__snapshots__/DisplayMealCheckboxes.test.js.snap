// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayMealCheckboxes component should match snapshot if error doesnt exists 1`] = `
<div>
  <b>
    Meals:
  </b>
  <div
    className="meals"
  />
</div>
`;

exports[`DisplayMealCheckboxes component should match snapshot if error exists 1`] = `
<div>
  <b>
    Meals:
  </b>
  <div
    className="meals error"
  />
  <div
    className="mealsErrorMessage"
  >
    error
  </div>
</div>
`;

exports[`DisplayMealCheckboxes component should match snapshot if meals are present 1`] = `
<div>
  <b>
    Meals:
  </b>
  <div
    className="meals error"
  >
    <div
      key="1"
    >
      <label
        className="checkBoxLabel"
      >
        rice
        <input
          checked="checked"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <div
      key="2"
    >
      <label
        className="checkBoxLabel"
      >
        beans
        <input
          checked=""
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
  </div>
  <div
    className="mealsErrorMessage"
  >
    error
  </div>
</div>
`;
